platform/flowglad-next/src/app/customers/[id]/purchases/columns.tsx

```diff
@@ -43,95 +49,104 @@ const PurchaseStatusCell = ({
 }
 
 export const columns: ColumnDef<PurchaseTableRowData>[] = [
+  {
+    id: 'purchaseDate',
+    accessorFn: (row) => row.purchase.purchaseDate,
+    header: 'Date',
+    size: 115,
+    minSize: 115,
+    maxSize: 144,
+    cell: ({ row }) => {
+      const date = row.getValue('purchaseDate') as Date | null
+      return <div>{date ? formatDate(date) : '-'}</div>
+    },
+  },
   {
     id: 'name',
     accessorFn: (row) => row.purchase.name,
-    header: 'Name',
-    cell: ({ row }) => (
-      <div className="truncate" title={row.getValue('name')}>
-        {row.getValue('name')}
-      </div>
-    ),
-    size: 300,
-    minSize: 150,
-    maxSize: 400,
+    header: 'Product',
+    size: 175,
+    minSize: 110,
+    maxSize: 250,
+    cell: ({ row }) => {
+      const name = row.getValue('name') as string
+      const purchaseId = row.original.purchase.id
+      return (
+        <DataTableLinkableCell
+          href={`/finance/purchases/${purchaseId}`}
+        >
+          <div className="truncate" title={name}>
+            {name}
+          </div>
+        </DataTableLinkableCell>
+      )
+    },
   },
   {
     id: 'status',
-    accessorFn: (row) => row.purchase,
+    accessorFn: (row) => getPurchaseStatusLabel(row.purchase),
     header: 'Status',
-    cell: ({ row }) => {
-      const purchase = row.getValue('status') as Purchase.ClientRecord
-      return <PurchaseStatusCell purchase={purchase} />
-    },
     size: 110,
-    minSize: 105,
-    maxSize: 115,
-  },
-  {
-    id: 'revenue',
-    accessorFn: (row) => row.revenue,
-    header: 'Revenue',
+    minSize: 110,
+    maxSize: 130,
     cell: ({ row }) => {
-      const revenue = row.getValue('revenue') as number | undefined
-      return (
-        <div className="whitespace-nowrap">
-          {stripeCurrencyAmountToHumanReadableCurrencyAmount(
-            CurrencyCode.USD,
-            revenue ?? 0
-          )}
-        </div>
-      )
+      const purchase = row.original.purchase
+      return <PurchaseStatusCell purchase={purchase} />
     },
-    size: 100,
-    minSize: 80,
-    maxSize: 120,
   },
   {
     id: 'customerName',
     accessorFn: (row) => row.customer.name || row.customer.email,
     header: 'Customer',
-    cell: ({ row }) => (
-      <div className="truncate" title={row.getValue('customerName')}>
-        {row.getValue('customerName')}
-      </div>
-    ),
-    size: 150,
-    minSize: 120,
-    maxSize: 200,
-  },
-  {
-    id: 'purchaseDate',
-    accessorFn: (row) => row.purchase.purchaseDate,
-    header: 'Purchase Date',
+    size: 165,
+    minSize: 113,
+    maxSize: 225,
     cell: ({ row }) => {
-      const date = row.getValue('purchaseDate') as Date | null
+      const original = row.original
+      const displayName =
+        original.customer.name.length === 0
+          ? original.customer.email
+          : original.customer.name
```

Comment:
Since empty strings are falsy we can simplify this check

```suggestion
       const displayName = 
          original.customer.name || 
          original.customer.email
```

Reply:
Thanks dude, fixed!